src/transaction/mod.rs

```diff
@@ -153,195 +114,56 @@ impl DbcTransaction {
-            .outputs
-            .iter()
-            .map(|out| RevealedOutput {
-                dbc_id: out.dbc_id,
-                revealed_amount: out.revealed_amount(&mut rng),
-            })
-            .take(self.outputs.len() - 1)
-            .collect();
-
-        // todo: replace fold() with sum() when supported in blstrs
-        let input_summed_blinding_factors: Scalar = revealed_input_amounts
-            .iter()
-            .map(RevealedAmount::blinding_factor)
-            .fold(Scalar::zero(), |sum, x| sum + x);
-
-        // todo: replace fold() with sum() when supported in blstrs
-        let output_summed_blinding_factors: Scalar = revealed_outputs
-            .iter()
-            .map(|r| r.revealed_amount.blinding_factor())
-            .fold(Scalar::zero(), |sum, x| sum + x);
-
-        let output_blinding_correction =
-            input_summed_blinding_factors - output_summed_blinding_factors;
-
-        if let Some(last_output) = self.outputs.last() {
-            revealed_outputs.push(RevealedOutput {
-                dbc_id: last_output.dbc_id,
-                revealed_amount: RevealedAmount {
-                    value: last_output.amount,
-                    blinding_factor: output_blinding_correction,
-                },
-            });
-        } else {
-            panic!("Expected at least one output")
-        }
-        revealed_outputs
-    }
-
-    fn blinded_outputs(
-        &self,
-        revealed_outputs: &[RevealedOutput],
-        mut rng: impl RngCore + CryptoRng,
-    ) -> Result<Vec<BlindedOutput>> {
-        let mut prover_ts = Transcript::new(MERLIN_TRANSCRIPT_LABEL);
-
-        let bp_gens = Self::bp_gens();
-
-        revealed_outputs
-            .iter()
-            .map(|c| {
-                let (range_proof, compressed_blinded_amount) = RangeProof::prove_single_with_rng(
-                    &bp_gens,
-                    &Self::pc_gens(),
-                    &mut prover_ts,
-                    c.revealed_amount.value,
-                    &c.revealed_amount.blinding_factor,
-                    RANGE_PROOF_BITS,
-                    &mut rng,
-                )?;
-                let blinded_amount = compressed_blinded_amount
-                    .decompress()
-                    .ok_or(Error::FailedToDecompressBlindedAmount)?;
-
-                Ok(BlindedOutput {
-                    dbc_id: c.dbc_id,
-                    range_proof,
-                    blinded_amount,
-                })
-            })
-            .collect::<Result<Vec<_>>>()
-    }
 }
 
-// note: used by both RevealedTx::sign and DbcTransaction::verify()
+// note: used by both TransactionIntermediate::sign and DbcTransaction::verify()
 //       which must match.
-fn serialize(
-    dbc_ids: &[DbcId],
-    input_amounts: &[BlindedAmount],
-    blinded_outputs: &[BlindedOutput],
-) -> Vec<u8> {
+fn serialize(dbc_ids: &[DbcId], input_amounts: &[Amount], outputs: &[Output]) -> Vec<u8> {
```

Comment:
I think me might not need a manual `serialize` for DBCs anymore as we don't have the un-serializable types from bulletproofs, we can just derive the trait.